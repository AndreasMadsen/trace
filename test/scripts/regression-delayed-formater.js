'use strict';

Error.prepareStackTrace = function (error, frames) {
  const lines = [ error.toString() ];
  for (let i = 0; i < frames.length; i++) {
    lines.push('    at-hack ' + frames[i].toString());
  }
  return lines.join('\n');
};

// Do something async-like
// Before fix this would result in a
// Cannot read property 'slice' of undefined
process.stdout.write('');

// Try getting a stack trace
throw new Error('trace');
