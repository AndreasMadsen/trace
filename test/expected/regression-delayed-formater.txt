/scripts/regression-delayed-formater.js:17
throw new Error('trace');
^

Error: trace
    at-hack Object.<anonymous> (/scripts/regression-delayed-formater.js:r:c)
    at-hack Module._compile (internal/modules/cjs/loader.js:r:c)
    at-hack Object.Module._extensions..js (internal/modules/cjs/loader.js:r:c)
    at-hack Module.load (internal/modules/cjs/loader.js:r:c)
    at-hack tryModuleLoad (internal/modules/cjs/loader.js:r:c)
    at-hack Function.Module._load (internal/modules/cjs/loader.js:r:c)
    at-hack Function.Module.runMain (internal/modules/cjs/loader.js:r:c)
    at-hack startup (internal/bootstrap/node.js:r:c)
    at-hack bootstrapNodeJSCore (internal/bootstrap/node.js:r:c)
